internal/controllers/reconciliation/controller.go

```diff
@@ -330,3 +329,33 @@ func mungePatch(patch []byte, rv string) ([]byte, error) {
 
 	return json.Marshal(patchMap)
 }
+
+// evalReadinessChecks evaluates and prioritizes readiness checks.
+// - Nil is returned when less than all of the checks are ready
+// - If some precise and some inprecise times are given, the precise times are favored
+// - Within precise or non-precise times, the max of that group is always used
```

Comment:
Could you elaborate a bit on precise vs non-precise times and how they're used?

Reply:
I see the code in api.go but I'd like to know how we're using it.

Reply:
Talked offline - essentially precise timestamps are derived from k8s condition transition times, non-precise are captured from the system time by the Eno controller.